blockchain/chain/src/store/chain_store.rs

```diff
@@ -107,19 +108,22 @@ where
             tipset_tracker: TipsetTracker::new(db.clone(), chain_config),
             db,
             ts_cache,
-            heaviest: Default::default(),
+            heaviest: Mutex::new(genesis_ts),
         };
 
         // Result intentionally ignored, doesn't matter if heaviest doesn't exist in store yet
         let _ = cs.load_heaviest_tipset();
 
-        cs
+        cs.blockstore().read(HEAD_KEY)?;
+        cs.blockstore().read(GENESIS_KEY)?;
```

Comment:
This does not do what you think it does. As you've written it, it just checks if there's a DB error while reading these keys. It doesn't check if those keys exists.

You need to make sure those keys exists. If they don't, make them exist. This will let you simplify the code that calls `ChainStore::new`. Make sure you understand why I'm asking you to do this before you do it.

Reply:
Hope that it is what is expected.